### ISSUE TITLE
Preserv json order

### ISSUE THREAD

**Issue by ErwanPigneul** [jmespath/jmespath.py#89]
Hello,

I'm not sur it's a bug but it would be nice if results would keep order of declaration :
"horaire[].{jour : jourOuverture.content, ouvertmatin : heureOuvertureAM.content, fermmatin : heureFermetureAM.content, ouvertaprem : heureOuverturePM.content, fermaprem : heureFermeturePM.content }"
won't keep order of attributes in python.
It can be { ouvertaprem : heureOuverturePM.content, jour : jourOuverture.content, ouvertmatin : heureOuvertureAM.content, fermaprem : heureFermeturePM.content, fermmatin : heureFermetureAM.content }

**Comment by jamesls** [comment]
One option would be to do what the `json.loads` function does, which is allow you to provide your own dictionary class to use when creating dict types.  So you can say:

```
json.loads(my_json_string, object_hook_pairs=collections.OrderedDict)
```

I could add support for:

```
jmespath.search(expression, data, dict_cls=collections.OrderedDict)
```

That would also provide more flexibility for people that want a different dict type, such as a sorted dict.

Marking as a feature request.

**Comment by ErwanPigneul** [comment]
Cool. Thanks. 

-----Message d'origine-----
De : "James Saryerwinnie" <notifications@github.com>
Envoyé : ‎23/‎09/‎2015 03:13
À : "jmespath/jmespath.py" <jmespath.py@noreply.github.com>
Cc : "Erwan Pigneul" <erwan.pigneul@gmail.com>
Objet : Re: [jmespath.py] Preserv json order (#89)

Closed #89 via #94.
